examples/src/test/kotlin/NetworkCompressionTest.kt

```diff
@@ -0,0 +1,99 @@
+import com.mongodb.*
+import com.mongodb.kotlin.client.coroutine.MongoClient
+import io.github.cdimascio.dotenv.dotenv
+import kotlinx.coroutines.runBlocking
+import org.bson.BsonInt64
+import org.bson.Document
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.TestInstance
+import java.util.*
+import kotlin.test.*
+
+// :replace-start: {
+//    "terms": {
+//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string>\"",
+//       "${uri}&": "mongodb+srv://<user>:<password>@<cluster-url>/?"
+//    }
+// }
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class NetworkCompressionTest {
+
+    companion object {
+        private val dotenv = dotenv()
+        val CONNECTION_URI_PLACEHOLDER = dotenv["MONGODB_CONNECTION_URI"]
+        var higherScopedClient: MongoClient? = null
+
+        @AfterAll
+        @JvmStatic
+        fun afterAll() {
+            runBlocking {
+                higherScopedClient?.close()
+            }
+        }
+
+    }
+
+    @Test
+    fun connectionStringCompressionTest() = runBlocking {
+        // :snippet-start: connection-string-compression-example
+        // Replace the placeholders with values from your connection string
+        val uri = CONNECTION_URI_PLACEHOLDER // :remove:
```

Comment:
rather than mark this to be removed, can we move it before the snippet starts, so it's not included at all?